src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -573,7 +572,7 @@ public Boolean call() throws Exception
 
             // failed to get connection from db, sleep and retry
             quietlySleep(sleepBackoff);
-            sleepBackoff = Math.min(connectionTimeout / 2, (long) (sleepBackoff * 1.3));
+            sleepBackoff = Math.min( TimeUnit.SECONDS.toMillis(5), Math.min(connectionTimeout, (long) (sleepBackoff * 1.3)));
```

Comment:
Am I missing something?  This is going to limit the backoff time to a maximum of 5 seconds?